scripts/pylib/twister/twisterlib.py

```diff
@@ -809,13 +809,19 @@ def run_custom_script(script, timeout):
     def handle(self):
         out_state = "failed"
         runner = None
+        timeout_cnt = self.timeout
```

Comment:
One personal opinion: I suggest a fixed period of time instead of the specify testsuite timeout.  Ex. 300 sec. Because this is a time of wait for getting the device.

Reply:
this timeout is set for each case,  so I reuse them.

Reply:
Maybe `jobs * timeout`? so that all parallel jobs can get  chance to be processed before timeout.

Reply:
Hi @hakehuang , I think this PR is good, it can solve in a scenario that the device cannot be make available and wait forever. But I still think this timeout should be a fixed period. My point is it's the time you are waiting for device available , this timeout usually depends on your environment, it seems like not related to the testcase itself. And a fixed timeout is more easily to adjust.